lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HadoopPaths.java

```diff
@@ -29,7 +30,7 @@ public static Path hadoopPath(Location location)
         // hack to preserve the original path for S3 if necessary
         String path = location.toString();
         Path hadoopPath = new Path(path);
-        if ("s3".equals(hadoopPath.toUri().getScheme()) && !path.equals(hadoopPath.toString())) {
```

Comment:
*suggestion*: maybe it would also be good to include alternative schemes in `TestS3HadoopPaths`?